waspc/cli/src/Wasp/Cli/Command/CreateNewProject/ProjectDescription.hs

```diff
@@ -0,0 +1,91 @@
+module Wasp.Cli.Command.CreateNewProject.ProjectDescription
+  ( initNewProjectDescription,
+    NewProjectDescription (..),
+  )
+where
+
+import Control.Monad (when)
+import Control.Monad.IO.Class (liftIO)
+import Data.List (intercalate)
+import Data.Maybe (isJust)
+import Path.IO (doesDirExist)
+import StrongPath.Path (toPathAbsDir)
+import Wasp.Analyzer.Parser (isValidWaspIdentifier)
+import Wasp.Cli.Command (Command)
+import Wasp.Cli.Command.Call (Arguments)
+import Wasp.Cli.Command.CreateNewProject.ArgumentsParser (NewProjectArgs (..), parseNewProjectArgs)
+import Wasp.Cli.Command.CreateNewProject.Common (getAbsoluteWaspProjectDir, throwInvalidTemplateNameUsedError, throwProjectCreationError)
+import Wasp.Cli.Command.CreateNewProject.Templates (StarterTemplateNames, getStarterTemplates, isValidTemplateName, templatesToList)
+import Wasp.Cli.Common (waspWarns)
+import qualified Wasp.Cli.Interactive as Interactive
+import Wasp.Cli.Terminal (asWaspWarningMessage)
+import Wasp.Util (indent, kebabToCamelCase)
+
+data NewProjectDescription = NewProjectDescription
+  { _projectName :: String,
+    _appName :: String,
+    _templateName :: Maybe String
+  }
+
+initNewProjectDescription :: Arguments -> Command NewProjectDescription
```

Comment:
Again me with the name, but I am not sure what to think about `init`, it sounds like it is initializing something, and I don't see what initialization means in this context, it is not like we have a multi-stage thing here.

How about `obtainNewProjectDescription`? Sounds like there is some work to be done here, with the obtaining, and we are doing some work to at the end give you new project description.

Reply:
`obtain` === `get` in my mind, and that's the name we had before. I've moved away from that on your suggestion that's not **only** getting stuff, but also doing something. So, `init` felt right since it's doing some prep work like parsing the args, getting the template names and then finally creating the `NewProjectDescription`

Reply:
I've ended up axing this function, I've just pulled the dependency fetching / preparing out.